taipy/gui/data/data_accessor.py

```diff
@@ -66,34 +74,59 @@ def to_csv(self, var_name: str, value: t.Any) -> t.Optional[str]:
 
 
 class _InvalidDataAccessor(_DataAccessor):
+
+    def __init__(self, gui: "Gui") -> None:
+        super().__init__(gui)
+
     @staticmethod
     def get_supported_classes() -> t.List[t.Type]:
         return []
 
     def get_data(
-        self, var_name: str, value: t.Any, payload: t.Dict[str, t.Any], data_format: _DataFormat
+        self,
+        var_name: str,
+        value: t.Any,
+        payload: t.Dict[str, t.Any],
+        data_format: _DataFormat,
     ) -> t.Dict[str, t.Any]:
+        transformed_value = self._gui.on_invalid_data(value)
```

Comment:
I think the callback should not be invoked here.
It should be invoked when evaluating the right `DataAccessor` for the data type.
this way, we can redirect to an existing `DataAccessor` and keep the `InvalidDataAccessor` for invalid data types.